airbyte-integrations/connectors/source-file/source_file/client.py

```diff
@@ -263,6 +263,10 @@ def load_nested_json_schema(self, fp) -> dict:
             builder.add_object(json.load(fp))
 
         result = builder.to_schema()
+        if "items" in result:
+            # this means we have a json list e.g. [{...}, {...}]
+            # but need to emit schema of an inside dict
+            result = result["items"]
```

Comment:
This looks good, but is there any thing to do / other validation if the json file is an array of anything other than object? 

For example, if the json file is just `["apple", "orange", "kiwi"]` this is probably invalid. (maybe a validation in `check` should be added here, but feel free to make a different issue for this and tackle it separately)

Reply:
seems to be valid
```
>>> builder = genson.SchemaBuilder()
>>> builder.add_object(['a', 'b', 'c'])
>>> builder.to_schema()["items"]
{'type': 'string'}
```

Reply:
Yeah the schema is technically valid for this, but what I mean is this will also cause issues with normalization since we expect each stream to have an object with properties as its schema (in this case it will just be a string). We probably shouldn't accept json files that aren't objects or an array of objects. Does that make sense?

Reply:
oh I see now
sure it makes sense, I will create a separate issue to work on this problem

Reply:
here it is https://github.com/airbytehq/airbyte/issues/16863